### PR TITLE
Further template propagation to fix offload build

### PR DESCRIPTION
**Pull request by PhilipFackler**

### CODE REVIEW

File: src/Numerics/OneDimGridFactory.h [hunk 1]
```diff
@@ -14,15 +14,17 @@
...
 #include "Numerics/OneDimGridFunctor.h"
+#include "Numerics/LibxmlNumericIO.h"
 
 namespace qmcplusplus
 {
 /** Factory class using Singleton pattern
  */
-struct OneDimGridFactory : public QMCTraits
+template <typename T>
+struct OneDimGridFactory
```
Comment (williamfgc) [inline src/Numerics/OneDimGridFactory.h:25]: Do we need to create a separate OneDimGridFactoryT ?

File: src/QMCWaveFunctions/ElectronGas/FreeOrbitalT.h [hunk 2]
```diff
@@ -22,66 +27,74 @@
...
+    {
+        throw std::runtime_error("not implemented");
+    }
+    // required overrides end ----
 private:
-  const std::vector<PosType> kvecs; // kvecs vectors
-  const int mink;                   // minimum k index
-  const int maxk;                   // maximum number of kvecs vectors
-  std::vector<RealType> k2neg;      // minus kvecs^2
+    const std::vector<PosType> kvecs; // kvecs vectors
```
Comment (williamfgc) [inline src/QMCWaveFunctions/ElectronGas/FreeOrbitalT.h:94]: @PhilipFackler let's make sure we are using a recent clang-format like clang-format-16 (that's what I am using).

File: src/Numerics/OneDimGridFactory.h (hunk 1 above)
Comment (williamfgc) [reply src/Numerics/OneDimGridFactory.h:25]: @PhilipFackler

**Review by williamfgc — CHANGES_REQUESTED** [review]

File: src/Numerics/OneDimGridFactory.h (hunk 1 above)
Comment (PhilipFackler) [reply src/Numerics/OneDimGridFactory.h:25]: No. It's only used in one place.

File: src/QMCWaveFunctions/ElectronGas/FreeOrbitalT.h (hunk 2 above)
Comment (PhilipFackler) [reply src/QMCWaveFunctions/ElectronGas/FreeOrbitalT.h:94]: Looks like I didn't format that file.

File: src/Numerics/OneDimGridFactory.h (hunk 1 above)
Comment (williamfgc) [reply src/Numerics/OneDimGridFactory.h:25]: OK, thanks for clarifying 👍

File: src/QMCWaveFunctions/ElectronGas/FreeOrbitalT.h (hunk 2 above)
Comment (williamfgc) [reply src/QMCWaveFunctions/ElectronGas/FreeOrbitalT.h:94]: Please check other files or an automatic way to format-on-save on your IDE. It pollutes the PR changes. Thanks!

Comment (PhilipFackler) [reply src/QMCWaveFunctions/ElectronGas/FreeOrbitalT.h:94]: Just kidding, I did format this file. And now I don't see the problem.

Comment (PhilipFackler) [reply src/QMCWaveFunctions/ElectronGas/FreeOrbitalT.h:94]: @williamfgc what would you like me to do about this?

Comment (williamfgc) [reply src/QMCWaveFunctions/ElectronGas/FreeOrbitalT.h:94]: Should be good now, thanks!

**Pull request merged**